pkg/controller/statusmanager/status_manager.go

```diff
@@ -186,6 +190,33 @@ func (status *StatusManager) SetDeployments(deployments []types.NamespacedName)
 func (status *StatusManager) SetRelatedObjects(relatedObjects []configv1.ObjectReference) {
 	status.Lock()
 	defer status.Unlock()
+
+	var found bool = false
+	for _, status_obj := range status.relatedObjects {
```

Comment:
You don't want to compare against the current value of `status.relatedObjects`, you want to compare against the `RelatedObjects` field of the actual operator status object. So you need to move all of this from `SetRelatedObjects` to `set`